finality-aleph/src/sync/handler/mod.rs

```diff
@@ -753,26 +754,39 @@ where
         let local_session = self
             .session_info
             .session_id_from_block_num(local_top_number);
-        let VerifiedHeader {
-            header,
-            maybe_equivocation_proof: maybe_proof,
-        } = self
-            .verifier
-            .verify_header(state.favourite_block(), false)
-            .map_err(Error::HeaderVerifier)?;
         let action = match local_session.0.checked_sub(remote_session.0) {
             // remote session number larger than ours, we can try to import the justification
-            None => HandleStateAction::maybe_extend(
-                self.handle_justification(state.top_justification(), Some(peer.clone()))?
-                    || self.forest.update_header(&header, Some(peer), false)?,
-            ),
+            None => {
+                let VerifiedHeader {
+                    header,
+                    maybe_equivocation_proof,
+                } = self
+                    .verifier
+                    .verify_header(state.favourite_block(), false)
+                    .map_err(Error::HeaderVerifier)?;
```

Comment:
nit: could we have a fn in handler `verify_header` which does exactly does -> call verify from verifier and maps an error

Reply:
done